src/gui/protocolitemmodel.cpp

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (C) by Hannah von Reth <hannah.vonreth@owncloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+#include "protocolitemmodel.h"
+
+#include "account.h"
+#include "accountstate.h"
+#include "models.h"
+#include "gui/folderman.h"
+
+#include "theme.h"
+
+#include <QIcon>
+
+namespace {
+auto getFolder(const OCC::ProtocolItem &item)
+{
+    auto f = OCC::FolderMan::instance()->folder(item.folderName());
+    OC_ASSERT(f);
+    return f;
+}
+}
+
+using namespace OCC;
+
+ProtocolItemModel::ProtocolItemModel(QObject *parent, bool issueMode, int maxLogSize)
+    : QAbstractTableModel(parent)
+    , _issueMode(issueMode)
+    , _maxLogSize(maxLogSize)
+{
+    _data.reserve(maxLogSize);
+}
+
+int ProtocolItemModel::rowCount(const QModelIndex &parent) const
+{
+    Q_ASSERT(checkIndex(parent));
+    if (parent.isValid()) {
+        return 0;
+    }
+    return actualSize();
+}
+
+int ProtocolItemModel::columnCount(const QModelIndex &parent) const
+{
+    Q_ASSERT(checkIndex(parent));
+    if (parent.isValid()) {
+        return 0;
+    }
+    return static_cast<int>(ProtocolItemRole::Count);
+}
+
+QVariant ProtocolItemModel::data(const QModelIndex &index, int role) const
```

Comment:
Wouldn't it make more sense to use `IdemDataRole role` instead? If you really need an `int` overload, you could add one that implements the `static_cast`.